recipes/generate.py

```diff
@@ -78,6 +78,30 @@ def generate(self, cfg: DictConfig):
         tokens = self._tokenizer.encode(cfg.prompt, add_bos=True, add_eos=False)
         prompt = torch.tensor(tokens, dtype=torch.int, device=self._device)
 
+        custom_generate_next_token = None
+
+        # since quantized model uses torch.compile to get speedup, it needs a warm up / prefill run
+        # to get the accurate performance measurement
+        if self._quantization_mode is not None:
+            t0 = time.perf_counter()
+            custom_generate_next_token = torch.compile(
+                utils.generate_next_token, mode="max-autotune", fullgraph=True
```

Comment:
noob question, what does this mode mean?

Reply:
just means flavors of compilation I think, here is the doc I found: https://pytorch.org/get-started/pytorch-2.0/#modes